string state
---
bool holding
string arm_index
